x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -229,6 +248,33 @@ export const lastValueOperation: OperationDefinition<LastValueIndexPatternColumn
       currentColumn.params.sortField,
       indexPattern
     );
+    const sourceIsScripted = isScriptedField(currentColumn.sourceField, indexPattern);
+
+    const usingTopValues = Object.keys(layer.columns).some(
+      (_columnId) => layer.columns[_columnId].operationType === 'terms'
+    );
+
+    const setShowArrayValues = (use: boolean) => {
+      let updatedLayer = updateColumnParam({
+        layer,
+        columnId,
+        paramName: 'showArrayValues',
+        value: use,
+      });
+
+      updatedLayer = {
+        ...updatedLayer,
+        columns: adjustColumnReferencesForChangedColumn(updatedLayer, columnId),
+      };
+
+      updateLayer(updatedLayer);
+    };
+
+    if (sourceIsScripted && !currentColumn.params.showArrayValues) {
+      // actively set this so that other places in the code know that we're using top-hit
+      setShowArrayValues(true);
```

Comment:
We should not manipulate state on render - this leads to hard to trace bugs because it makes a difference whether the user opens the dimension editor or not (e.g. by dragging in a field). Let's invoke this logic in the `onFieldChange` callback instead.

Reply:
Very good catch

Reply:
Actually, though, `onFieldChange` is only called with the field changes. Not when the operation is initially selected. So, if I drag a scripted field to the metric landing and then switch to last_value, this won't run.

https://user-images.githubusercontent.com/315764/157475074-2355e2ca-09d2-48e5-9c1e-8d666be3b006.mov

Reply:
@andrewctate You should be able to do the same check in `buildColumn` as well (which is called on drop)